File: pinyougou/pinyougou-seckill/pinyougou-seckill-interface/src/main/java/com/pinyougou/seckill/service/SeckillOrderService.java
package com.pinyougou.seckill.service;

import com.pinyougou.pojo.TbOrderItem;
import com.pinyougou.pojo.TbSeckillGoods;
import com.pinyougou.pojo.TbSeckillOrder;
import com.pinyougou.service.BaseService;
import com.pinyougou.vo.PageResult;

import java.util.List;

public interface SeckillOrderService extends BaseService<TbSeckillOrder> {

    PageResult search(Integer page, Integer rows, TbSeckillOrder seckillOrder);

    /**
     * 生成秒杀订单并返回id
     * @param username
     * @param seckillId
     * @return
     */
    Long submitOrder(String username, Long seckillId) throws Exception;

    /**
     * 根据订单 id 查询放置在 redis 中的订单
     * @param outTradeNo
     * @return
     */
    TbSeckillOrder getSeckillOrderInRedisByOrderId(String outTradeNo);

    /**
     * 更新redis中的支付状态,并保存到数据库中
     * @param outTradeNo
     * @param transactionId
     */
    void updateOrderInRedisStatus(String outTradeNo, String transactionId);

    /**
     * 删除redis中的订单
     * @param outTradeNo
     */
    void deleteOrderInRedisByIds(String outTradeNo);
}

File: pinyougou/pinyougou-shop-web/src/main/webapp/js/service/seckillorderService.js
app.service("seckillorderService", function ($http) {
    //加载列表数据
    this.findAll = function () {
        return $http.get("../seckillorder/findAll.do");
    };

    this.findPage = function (page, rows) {
        return $http.get("../seckillorder/findPage.do?page=" + page + "&rows=" + rows);
    };

    this.update = function (entity,id) {
        return $http.post("../seckillorder/update.do?id="+id, entity);
    };

    this.findOne = function (id) {
        return $http.get("../seckillorder/findOne.do?id=" + id);
    };

    this.search = function (page, rows, searchEntity) {
        return $http.post("../seckillorder/search.do?page=" + page + "&rows=" + rows, searchEntity);

    };

    this.findAllOrder = function () {
        return $http.post("../seckillorder/findAllOrder.do");
    };
});